sdk/Sdk/Tasks/EnrichExtensionsMetadata.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Tasks
+{
+    public class EnrichExtensionsMetadata : Task
+    {
+        [Required]
+        public string? ExtensionsJsonPath { get; set; }
+
+        [Required]
+        public string? OutputPath { get; set; }
+
+        public override bool Execute()
+        {
+            string json = File.ReadAllText(ExtensionsJsonPath);
+
+            var extensions = JsonSerializer.Deserialize<ExtensionsMetadata>(json)?.Extensions ?? Enumerable.Empty<ExtensionReference>();
+            ExtensionsMetadataEnhancer.AddHintPath(extensions);
+
+            string newJson = GenerateExtensionsJson(extensions);
+            File.WriteAllText(OutputPath, newJson);
+
+            File.Delete(ExtensionsJsonPath);
+
+            return true;
+        }
+
+        // From ExtensionsMetadataGenerator
+        // https://github.com/Azure/azure-functions-host/blob/51e7a2ead2327889da27c28681f89e60bd75a46f/tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs#L71
+        private static string GenerateExtensionsJson(IEnumerable<ExtensionReference> extensionReferences)
+        {
+            var referenceObjects = extensionReferences.Select(r => string.Format("{3}    {{ \"name\": \"{0}\", \"typeName\":\"{1}\", \"hintPath\":\"{2}\"}}", r.Name, r.TypeName, r.HintPath, Environment.NewLine));
+            string json = string.Format("{{{1}  \"extensions\":[{0}{1}  ]{1}}}", string.Join(",", referenceObjects), Environment.NewLine);
+            return json;
+        }
```

Comment:
Why'd you have to do your own string formatting? Could you let JSON Serialization handle all this for you?

Reply:
I don't need to. But I just copied the logic for how we write in ExtensionsMetadataGenerator ([here](https://github.com/Azure/azure-functions-host/blob/51e7a2ead2327889da27c28681f89e60bd75a46f/tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs#L71)) to avoid disparity. Had added a comment [here](https://github.com/Azure/azure-functions-dotnet-worker/pull/119/files/a105f1ee09b2e091dc2fdf544e0251169af07d07#diff-2a1b9b73617945b8cb93b0719e4b1d78fe442fda27784c48e5e593bd86c2897eR36). But yeah, there shouldn't be any disparity anyway, it's just a JSON file.

I will update to use the JsonWriter.